client/gutenberg/extensions/tiled-gallery/edit.jsx

```diff
@@ -0,0 +1,230 @@
+/** @format */
+
+/**
+ * External Dependencies
+ */
+import pick from 'lodash/pick';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component, Fragment } from '@wordpress/element';
+import {
+	BlockControls,
+	InspectorControls,
+	MediaPlaceholder,
+	MediaUpload,
+	mediaUpload,
+} from '@wordpress/editor';
+import {
+	DropZone,
+	IconButton,
+	PanelBody,
+	RangeControl,
+	SelectControl,
+	Toolbar,
+	withNotices,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import TiledGallerySave from './save.jsx';
+
+/**
+ * Module variables
+ */
+const MAX_COLUMNS = 8;
+const linkOptions = [
+	{ value: 'attachment', label: __( 'Attachment Page' ) },
+	{ value: 'media', label: __( 'Media File' ) },
+	{ value: 'none', label: __( 'None' ) },
+];
+
+class TiledGalleryEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.onSelectImage = this.onSelectImage.bind( this );
+		this.onSelectImages = this.onSelectImages.bind( this );
+		this.setLinkTo = this.setLinkTo.bind( this );
+		this.setColumnsNumber = this.setColumnsNumber.bind( this );
+		this.setImageAttributes = this.setImageAttributes.bind( this );
+		this.addFiles = this.addFiles.bind( this );
+
+		this.state = {
+			selectedImage: null,
+		};
```

Comment:
Is this constructor declaration following Gutenberg style? Most of Calypso wouldn't use a constructor for any of this.

Reply:
Seems to be a direct copy from Core gallery: https://github.com/WordPress/gutenberg/blob/2b42db51dd1a6c100d22190669a14a7873af6a44/packages/block-library/src/gallery/edit.js#L60-L62 (lots of the code here seems to be).

I'll make sure to Calypso'fy in next iteration. :thumbsup: